### PR TITLE
[G7T5-9 & G7T5-197] HR Add and Remove Skills from Job Role

### PR DESCRIPTION
**Pull request by deadline2359**
# Description

Fixes [Jira ticket G7T5-9](https://is212g7t5.atlassian.net/browse/G7T5-9) and [Jira ticket G7T5-197](https://is212g7t5.atlassian.net/browse/G7T5-197)

https://user-images.githubusercontent.com/89028059/200041428-a790a764-3986-4bc7-b468-316c25cd5e94.mov

List all acceptance criterias of user story and indicate which has been solved previously and which is solved currently
**Jira ticket G7T5-9**
- [x] HR can only select 1 Active Job Role at once
- [x] HR cannot select Inactive Job Roles
- [x] Upon selection of the Job Role,
    - [x] HR can view and select from a list of only Active Skills
    - [x] Skills that are already assigned to the Job Role must be differentiated
- [x] HR can assign skill(s) to the selected Job Role
    - [x] Assigned skill(s) must be active
    - [x] Selected Job Role must be active
- [x] HR cannot re-assign a skill already assigned to the Job Role. 
- [x] Skill(s) can be assigned to Job Role on both desktop and mobile devices
- [x] Skill(s) can be assigned to Job Role in less than 3 seconds

**Jira ticket G7T5-197**
- [x] HR can select a Job Role from a list of Job Roles
- [x] HR can view a list of Skills corresponding to said Job Role
    - [x] The list of Skills should accurately reflect, if any, the Skill(s) currently assigned to the Job Role.
- [x] HR can remove Skill(s) from the selected Job Role 
- [x] Removal of Skill from Job Role should take less than 3 seconds
- [x] HR can remove Skills on both mobile and desktop devices
- [x] Staff with existing Learning Journeys of Job Roles that have these deleted skills:
    - [x] Will no longer see the deleted Skill(s) in the Learning Journeys

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor
- [ ] Migration

# How Has This Been Tested?
- [x] I have added the corresponding testcases into the [testcase document](https://docs.google.com/spreadsheets/d/1QOyUN_kFN0fddLkjAQz2DK3jou3cWdN9CJsNbl3v0Kg/edit#gid=0)

Please indicate the testcase ID you have added or are using

**G7T5-9**
- [x] G7T5-9-1
- [x] G7T5-9-2
- [x] G7T5-9-3
- [x] G7T5-9-4
- [x] G7T5-9-5
- [x] G7T5-9-6
- [x] G7T5-9-7
- [x] G7T5-9-8
- [x] G7T5-9-9
- [x] G7T5-9-10
- [x] G7T5-9-11
- [x] G7T5-9-12

**G7T5-197**
- [x] G7T5-18-1
- [x] G7T5-18-2
- [x] G7T5-18-3
- [x] G7T5-18-4
- [x] G7T5-18-5
- [x] G7T5-18-6
- [x] G7T5-18-7
- [x] G7T5-18-8

# Checklist:

- [x] I have added the appropriate labels to my PR
- [x] My code follows the style guidelines of this project
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/src/components/job/hr/CreateJob.jsx [hunk 1]
```diff
@@ -26,6 +29,9 @@ export default function HRCreateJob() {
       setErrors(errorList);
     } else {
       setErrors([]);
+      selectedSkills.map(async (skillId) => {
```
Comment (quinncheong) [inline app/src/components/job/hr/CreateJob.jsx:32]: The use of map should be to return an array. Use a for loop if you want to do actions or else the event loop will bug out

File: app/src/components/job/hr/CreateJob.jsx [hunk 2]
```diff
@@ -36,6 +42,7 @@ export default function HRCreateJob() {
     setJobName("");
     setJobDesc("");
     setErrors([]);
```
Comment (quinncheong) [inline app/src/components/job/hr/CreateJob.jsx:44]: If seterrors is here and it is the final call, not sure if the errors will forever be empty.

File: app/src/components/job/hr/UpdateJobSuccess.jsx [hunk 3]
```diff
@@ -1,29 +1,60 @@
...
-export default function UpdateJobSuccess({ jobName }) {
+export default function UpdateJobSuccess({ jobName, jobIsActive }) {
   const history = useHistory();
 
   const redirectToJobPage = () => {
     history.push("/jobs");
   };
 
   return (
+    // <div
```
Comment (quinncheong) [inline app/src/components/job/hr/UpdateJobSuccess.jsx:12]: remove comments

**Review by quinncheong — CHANGES_REQUESTED** [review]

File: app/src/components/job/hr/CreateJob.jsx (hunk 2 above)
Comment (deadline2359) [reply app/src/components/job/hr/CreateJob.jsx:44]: image

I don't think so. I tried a few times and seemed to work well. This particular `setError()` only passes to clear previous errors when the user creates a new job, after already created one.

File: app/src/components/job/hr/CreateJobSuccess.jsx [hunk 4]
```diff
@@ -14,28 +13,35 @@ export default function CreateJobSuccess({ jobName, resetFields }) {
...
+      <div className='flex-initial'>
+        <div className='container shadow-lg px-7 py-5 grid rounded-lg bg-white'>
+          <div className='grid-row py-3 text-3xl font-bold'>Note</div>
+
+          <div className='grid-row py-3 text-lg'>{jobName} has been successfully created!</div>
+
+          <div className='grid-row py-3 flex justify-end'>
+            <button
+              type='button'
+              className='text-white bg-accent1 hover:bg-accent2 font-medium rounded-lg text-sm px-5 py-2.5 mr-2 mb-2'
```
Comment (theianchia) [inline app/src/components/job/hr/CreateJobSuccess.jsx:30]: shld be `bg-accent2` to `hover:bg-accent3`

**Review by theianchia — CHANGES_REQUESTED** [review]

**Review by quinncheong — APPROVED** [review]

**Review by theianchia — APPROVED** [review]

**Pull request merged**